pkg/asset/agent/mirror/registriesconf.go

```diff
@@ -45,32 +126,44 @@ func (*RegistriesConf) Dependencies() []asset.Asset {
 
 // Generate generates the registries.conf file from install-config.
 func (i *RegistriesConf) Generate(dependencies asset.Parents) error {
+	installConfig := &agent.OptionalInstallConfig{}
+	dependencies.Get(installConfig)
+	if !installConfig.Supplied || len(installConfig.Config.ImageContentSources) == 0 {
```

Comment:
Actually I don't think we can add the default registries.conf file, at least not in its current form. Assisted-service isn't able to process it because it is expecting the [registry] table to be defined if the file is present: https://github.com/openshift/assisted-service/blob/master/pkg/mirrorregistries/generator.go#L69  

Consequently the cluster fails to install and I am seeing this in the event log:
![Screen Shot 2022-07-11 at 11 13 20 PM](https://user-images.githubusercontent.com/974403/178299205-8cc55303-1dec-41fd-b50e-422d569738da.png)

Reply:
I wonder if just uncommenting the 
`[[registry]]`
and 
`[[registry.mirror]]`

in the default file would allow to be considered as as valid TOML. Otherwise we'd need to avoid creating an asset from this default file if there is no valid registries info in the installConfig

Reply:
It looks like they also require location fields to be set. We should probably fix this in assisted-service in the long term. Just not creating the file in ignition won't help, because the default in CoreOS also has no mirrors configured (in fact it's essentially this same file). So we will have to make the mount conditional for now. Trying that out at the moment.

Reply:
Sorted. Thanks Richard!